### PR TITLE
add ensure deceased to survival time field

### PR DESCRIPTION
**Pull request by wajiha-oicr**
- updated the ensure deceased validation script to use the name parameter
- added the script to the survival_time field
- updated the unit tests accordingly

### CODE REVIEW

File: schemas/donor.json [hunk 1]
```diff
@@ -67,6 +67,9 @@
         "validationDependency": true,
         "units": "days",
         "core": "true"
+      },
+      "restrictions": {
+        "script": "#/script/donor/ensureDeceased"
```
Comment (rosibaj) [inline schemas/donor.json]: can we add the meta field updates as well
    
``` "meta": {	        
      "dependsOn": "donor.vital_status",
        "notes": "Survival_time is only required to be submitted if the donor's vital_status is Deceased."	       
      }	      
}
```

**Comment by wajiha-oicr** [comment]
@joneubank  can I get a review?

File: schemas/donor.json [hunk 2]
```diff
@@ -64,9 +64,14 @@
       "name": "survival_time",
       "valueType": "integer",
       "meta": {
+        "dependsOn": "donor.vital_status",
```
Comment (joneubank) [inline schemas/donor.json]: 👍

**Comment by wajiha-oicr** [comment]
@rosibaj updated dictionary is in dev and qa

**Pull request merged**